### PR TITLE
docstring function.py

### PR DESCRIPTION
**Pull request by D1792**
please review

### CODE REVIEW

**Review by tguens — COMMENTED** [review]
No need to say that. (no need to sya what's not implemented). Just say base class that implements the chain rule operations for autodiff.Variable and functions.

File: autodiff/function.py [hunk 1]
```diff
@@ -17,58 +17,105 @@
 """
 
 class Function:
+    """
+   The get_grad and get_val methods are not implemented for this base class
```
Comment (tguens) [inline autodiff/function.py:19]: My comment to be inserted here

File: autodiff/function.py [hunk 2]
```diff
@@ -17,58 +17,105 @@
...
     def get_grad(self, x):
-        #Works on array
+        """Implements calculation of derivative
+    
+        INPUTS
+        =======
+        x: numpy.array
+            The points at which we are evalutating the derivative
+    
+        RETURNS
```
Comment (tguens) [inline autodiff/function.py]: it's not implemented !!!! NO INPUT OR OUTPUT.

File: autodiff/function.py [hunk 3]
```diff
@@ -17,58 +17,105 @@
...
+        ========
+        numpy.array: The derivative of corresponding elementary function.
+        
+        """
         raise NotImplementedError
 
     def get_val(self, x):
-        #Works on array
+        """Implements calculation of value
```
Comment (tguens) [inline autodiff/function.py]: SAME THING. not implemented.

File: autodiff/function.py [hunk 4]
```diff
@@ -17,58 +17,105 @@
...
+        """
         return '{}'.format(type(self))
 
-    #Works on AD.Variable
     def __call__(self, x):
-        """
-        Implements the chain rule.
-        Input: autodiff.Variable type holding a val and grad
-        Output:  autodiff.Variable type holding the val, grad of the transormed variable
+        """Implements the chain rule.
```
Comment (tguens) [inline autodiff/function.py:42]: Good.

File: autodiff/function.py [hunk 5]
```diff
@@ -17,58 +17,105 @@
...
         """
         out_val = self.get_val(x.val)
         out_grad = np.dot(self.get_grad(x.val), x.grad)
         return Variable(val=out_val, grad=out_grad)
     
 class Exponent(Function):
-    """Exponential"""    
-    def get_val(self, x):
+        """Implements calculation of value and derivative of Exponential function
+        Overloads get_val and get_grad from the Function class
```
Comment (tguens) [inline autodiff/function.py:58]: yes

File: autodiff/function.py [hunk 6]
```diff
@@ -17,58 +17,105 @@
 """
```
Comment (tguens) [inline autodiff/function.py:15]: Please can you wirte a general doc that explains that functions transforms variable to variable AND that currently only th following functoins are implemented (exp, sin,...)

**Review by Josh-Bodner — APPROVED** [review]
Looks good!

**Pull request merged**